front/app/services/notifications.ts

```diff
@@ -382,46 +382,40 @@ export interface IThresholdReachedForAdminNotificationData
-  | IStatusChangeOnCommentedIdeaNotificationData
-  | IStatusChangeOnCommentedInitiativeNotificationData
-  | IStatusChangeOnVotedIdeaNotificationData
-  | IStatusChangeOnVotedInitiativeNotificationData
-  | IThresholdReachedForAdminNotificationData
-  | INlpFlaggedPostNotificationData;
+export interface INotificationDataMap {
+  IAdminRightsReceivedNotificationData: IAdminRightsReceivedNotificationData;
+  ICommentDeletedByAdminNotificationData: ICommentDeletedByAdminNotificationData;
+  ICommentMarkedAsSpamNotificationData: ICommentMarkedAsSpamNotificationData;
+  ICommentOnYourCommentNotificationData: ICommentOnYourCommentNotificationData;
+  ICommentOnYourIdeaNotificationData: ICommentOnYourIdeaNotificationData;
+  ICommentOnYourInitiativeNotificationData: ICommentOnYourInitiativeNotificationData;
+  IIdeaAssignedToYouNotificationData: IIdeaAssignedToYouNotificationData;
+  IIdeaMarkedAsSpamNotificationData: IIdeaMarkedAsSpamNotificationData;
+  IInitiativeAssignedToYouNotificationData: IInitiativeAssignedToYouNotificationData;
+  IInitiativeMarkedAsSpamNotificationData: IInitiativeMarkedAsSpamNotificationData;
+  IInviteAcceptedNotificationData: IInviteAcceptedNotificationData;
+  IMentionInCommentNotificationData: IMentionInCommentNotificationData;
+  IMentionInOfficialFeedbackNotificationData: IMentionInOfficialFeedbackNotificationData;
+  IOfficialFeedbackOnCommentedIdeaNotificationData: IOfficialFeedbackOnCommentedIdeaNotificationData;
+  IOfficialFeedbackOnCommentedInitiativeNotificationData: IOfficialFeedbackOnCommentedInitiativeNotificationData;
+  IOfficialFeedbackOnVotedIdeaNotificationData: IOfficialFeedbackOnVotedIdeaNotificationData;
+  IOfficialFeedbackOnVotedInitiativeNotificationData: IOfficialFeedbackOnVotedInitiativeNotificationData;
+  IOfficialFeedbackOnYourIdeaNotificationData: IOfficialFeedbackOnYourIdeaNotificationData;
+  IOfficialFeedbackOnYourInitiativeNotificationData: IOfficialFeedbackOnYourInitiativeNotificationData;
+  IProjectModerationRightsReceivedNotificationData: IProjectModerationRightsReceivedNotificationData;
+  IProjectPhaseStartedNotificationData: IProjectPhaseStartedNotificationData;
+  IProjectPhaseUpcomingNotificationData: IProjectPhaseUpcomingNotificationData;
+  IStatusChangeOfYourIdeaNotificationData: IStatusChangeOfYourIdeaNotificationData;
+  IStatusChangeOfYourInitiativeNotificationData: IStatusChangeOfYourInitiativeNotificationData;
+  IStatusChangeOnCommentedIdeaNotificationData: IStatusChangeOnCommentedIdeaNotificationData;
+  IStatusChangeOnCommentedInitiativeNotificationData: IStatusChangeOnCommentedInitiativeNotificationData;
+  IStatusChangeOnVotedIdeaNotificationData: IStatusChangeOnVotedIdeaNotificationData;
+  IStatusChangeOnVotedInitiativeNotificationData: IStatusChangeOnVotedInitiativeNotificationData;
+  IThresholdReachedForAdminNotificationData: IThresholdReachedForAdminNotificationData;
+}
+
+export type TNotificationData = INotificationDataMap[keyof INotificationDataMap];
```

Comment:
Making it possible to extend TNotificationData from outside this file.

Reply:
This is really well typed 👍